protocol/0019-MCAL-margin_calculator.md

```diff
@@ -40,23 +40,98 @@
+
+1. **Isolated margin mode**: this mode sacrifices capital efficiency for predictability and risk management by segregating positions.
+In this mode, the entire margin for any newly opened position volume is transferred to the margin account when the trade is executed. 
+This includes completely new positions and increases to position size.
+
+
+### Actual position margin levels:
+
+1. **Maintenance margin**: the minimum margin a party must have in their margin account to avoid the position being liquidated.
+
+1. **Collateral search level**: when in cross-margin mode, the margin account balance below which the system will seek to recollateralise the margin account back to the initial margin level.
+
+1. **Initial margin**: when in cross-margin mode, the margin account balance initially allocated for the position, and the balance to which the margin account will be returned after collateral search and release, if possible.
+
+1. **Collateral release level**: when in cross-margin mode, the margin account balance above which the system will return collateral from a profitable position to the party's general account for potential use elsewhere.
+
+
+It is always the case that:
+
+```
+maintenance margin < collateral search level < initial margin < collateral release level
+```
+
+
+### Potential position margin level:
+
+1. **Order margin**: the amouht of additional margin on top of the amount in the margin account that is required for the party's current active orders.
+Note that this may be zero if the active orders can only decrease the position size. 
 
-1. ***Maintenance margin***
-1. ***Collateral search level***
-1. ***Initial margin***
-1. ***Collateral release level***
 
-The protocol is designed such that ***Maintenance margin < Collateral search level < Initial margin < Collateral release level***.
 
 Margin levels are used by the protocol to ascertain whether a trader has sufficient collateral to maintain a margined trade. When the trader enters an open position, this required amount is equal to the *initial margin*. Subsequently, throughout the life of this open position, the minimum required amount is the *maintenance margin*. As a trader's collateral level dips below the *collateral search level* the protocol will automatically search for more collateral to be assigned to support this open position from the trader's general collateral accounts. In the event that a trader has collateral that is above the *collateral release level* the protocol will automatically release collateral to a trader's general collateral account for the relevant asset.
 
 **Whitepaper reference:** 6.1, section "Margin Calculation"
 
 In future there can be multiple margin calculator implementations that would be configurable in the market framework. This spec describes one implementation.
 
+
+
+## Isolated margin mode
```

Comment:
I'm perhaps missing the point, but could we not just implement margin segregation as user-initiated transfers between their general and margin account?

So when margin segregation is flipped on for a market we never search or release collateral. User can always transfer into margin account. They can only transfer from margin into general if the amount left after transfer is at or above the margin release level at that time (this could be a bit tricky as levels change all the time, but maybe we can just allow users to transfer out max allowed at the time of transfer - actual amount is never specified). 

So if they want to say significantly increase their position they'd first need to transfer enough funds into the margin account for that market, and then open then place order as usual.

Those transfers could be free or very cheap (time-throttled?) if we worry about spam.

Reply:
@barnabee / @TomMcL that sounds reasonable, what do you think?